apdx-validation.tex

```diff
@@ -78,88 +83,243 @@ \subsubsection*{Rules for SBOL Objects}
+\sbol{Interaction} 	& \sbol{Identified} 	& \sbolmult{type:I}{type}, \sbol{hasParticipation} & \sec{sec:Interaction} \\
+\sbol{Interface} 		& \sbol{Identified} 	& \sbol{input}, \sbol{output}, \sbol{nondirectional} & \sec{sec:Interface} \\
+\sbol{LocalSubComponent} & \sbol{Feature} & \sbolmult{type:LSC}{type}, \sbolmult{hasLocation:LSC}{hasLocation} & \sec{sec:LocalSubComponent} \\
+\sbol{Location}		& \sbol{Identified} 	& \sbolmult{orientation:L}{orientation}, \sbol{order}, \sbolmult{hasSequence:L}{hasSequence} & \sec{sec:Location} \\
+\sbol{Model} 		& \sbol{TopLevel} 	& \sbolmult{source:M}{source}, \sbol{language}, \sbol{framework} & \sec{sec:Model} \\
+\sbol{Participation} 	& \sbol{Identified} 	& \sbolmult{role:P}{role}, \sbol{participant}, \sbol{higherOrderParticipant} & \sec{sec:Participation} \\
+\sbol{Range} 		& \sbol{Location} 	& \sbol{start}, \sbol{end} & \sec{sec:Range} \\
+\sbol{SequenceFeature} & \sbol{Feature}	& \sbolmult{hasLocation:SF}{hasLocation} & \sec{sec:SequenceFeature} \\
+\sbol{Sequence} 	& \sbol{TopLevel} 	& \sbol{elements}, \sbol{encoding} & \sec{sec:Sequence} \\
+\sbol{SubComponent} & \sbol{Feature} 	& \sbol{roleIntegration}, \sbol{instanceOf}, \sbol{sourceLocation}, \sbolmult{hasLocation:SC}{hasLocation} & \sec{sec:SubComponent} \\
+\sbol{TopLevel}		& \sbol{Identified} 	& \sbol{hasNamespace}, \sbol{hasAttachment} & \sec{sec:TopLevel} \\
+\sbol{VariableFeature} & \sbol{Identified} 	& \sbol{cardinality}, \sbol{variable}, \sbol{variant}, \sbol{variantCollection}, \sbol{variantDerivation}, \sbol{variantMeasure} & \sec{sec:VariableFeature} \\
+\hline
+\prov{Activity}		& \sbol{TopLevel}	&	& \sec{sec:prov:Activity}\\
+\prov{Agent}		& \sbol{TopLevel}	&	& \sec{sec:prov:Agent}\\
+\prov{Association}	& \sbol{Identified} 	&	& \sec{sec:prov:Association}\\
+\prov{Plan}		& \sbol{TopLevel}	&	& \sec{sec:prov:Plan}\\
+\prov{Usage}		& \sbol{Identified}	&	& \sec{sec:prov:Usage}\\
+\hline
+\om{BinaryPrefix}	& \om{Prefix}		& 	& \sec{sec:om:BinaryPrefix}\\
+\om{CompoundUnit}	& \om{Unit}		& 	& \sec{sec:om:CompoundUnit}\\
+\om{Measure}		& \sbol{Identified}	& \sbolmult{type:Measure}{type} & \sec{sec:om:Measure}\\
+\om{PrefixedUnit}	& \om{Unit}		& 	& \sec{sec:om:PrefixedUnit}\\
+\om{Prefix}		& \sbol{TopLevel}	& 	& \sec{sec:om:Prefix}\\
+\om{SIPrefix}		& \om{Prefix}		& 	& \sec{sec:om:SIPrefix}\\
+\om{SingularUnit}	& \om{Unit}		& 	& \sec{sec:om:SingularUnit}\\
+\om{UnitDivision} 	& \om{CompoundUnit} & & \sec{sec:om:UnitDivision}\\
+\om{UnitExponentiation} & \om{CompoundUnit} & & \sec{sec:om:UnitExponentiation}\\
+\om{UnitMultiplication} & \om{CompoundUnit} & & \sec{sec:om:UnitMultiplication}\\
+\om{Unit}			& \sbol{TopLevel}	& 	& \sec{sec:om:Unit}\\
+\hline
+\end{longtable}
+\end{scriptsize}
+
+\begin{scriptsize}
+\setlength{\tabcolsep}{1pt}
+\begin{longtable}{|llllll|}
+\caption{Cardinality constraints on object properties, their types, and types of referred objects.\label{t:property_cardinality_type}}\\
```

Comment:
I am having trouble understanding the terms in this column and their various permutations, e.g., `REQUIRED`, `REQUIRED, ONE`, `REQUIRED, UNBOUNDED`.  I suggest adding definitions for these terms to the table caption, or alternatively, use the following terms, which I believe is easier to understand when using this table as a quick reference: `ExactlyOne`, `OneOrMore`, `ZeroOrOne`, `ZeroOrMore`. (These are also terms defined in RDF schema)

Reply:
These terms are taken from Section 4.2 of the specification.  If you'd like to change them, I would suggest opening a separate issue that suggests a change to both Section 4.2 and this table.

Reply:
In the table caption, can we add a reference pointer back to Section 4.2 where the terms are defined?

What is particularly confusing is the combination of these terms:
`REQUIRED`
`REQUIRED, ONE`
`REQUIRED, UNBOUNDED`

From context alone, when reading the table, the meaning of unqualified `REQUIRED` is ambiguous -- does it imply `REQUIRED, ONE` or `REQUIRED, UNBOUNDED`, or something different?   

To be consistent with the terms defined in sec. 4.2, I would recommend eliminating `REQUIRED, ONE`

Reply:
Oh, I see the problem.  I wrote the SBOL section with "REQUIRED, ONE" because that's what's one page 11 line 14 in section 4.2.  But then I forgot the "ONE" in the PROV and OM sections.  I will switch it all to "REQUIRED, ONE" because that's what's in Section 4.2.